oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/oidc/authentication/OidcClientRegistrationAuthenticationProvider.java

```diff
@@ -183,7 +195,26 @@ private OidcClientRegistrationAuthenticationToken registerClient(OidcClientRegis
 		}
 
 		RegisteredClient registeredClient = this.registeredClientConverter.convert(clientRegistrationAuthentication.getClientRegistration());
-		this.registeredClientRepository.save(registeredClient);
+
+		// Copy RegisteredClient and encode only secret
+		String rawClientSecret = registeredClient.getClientSecret();
```

Comment:
The extension point for customizing client metadata before it's saved to `RegisteredClientRepository` is `OidcClientRegistrationAuthenticationProvider.setRegisteredClientConverter()`. It's not currently documented in the reference manual but there is an open issue gh-1044 that provides sample code on how to configure.